dotnet/src/InternalUtilities/src/Schema/JsonSchemaMapper.cs

```diff
@@ -321,12 +321,8 @@ public static string ToJsonString(this JsonNode? node, bool writeIndented = fals
                 {
                     if (TryGetStringEnumConverterValues(typeInfo, effectiveConverter, out JsonArray? values))
                     {
-                        if (values is null)
-                        {
-                            // enum declared with the flags attribute -- do not surface enum values in the JSON schema.
-                            schemaType = JsonSchemaType.String;
-                        }
-                        else
+                        schemaType = JsonSchemaType.String;
+                        if (values != null)
```

Comment:
I'm looking here: https://json-schema.org/understanding-json-schema/reference/enum

Which says _You can use enum even without a type, to accept values of different types._

Could the bug be in the Google service? Can you check to see has this been reported as a issue.

You mention things work fine with OpenAI. I just tried MistralAI and it works with and without the type.

Reply:
Maybe this is limitation Google rest api. It always requires specify type.

Reply:
@markwallace-microsoft I think adding type is good solution, it doesn't break other connectors and work with gemini, JsonSchemaMapper uses `TryGetStringEnumConverterValues` which internally adds values as strings
```cs
var namingPolicy = (JsonNamingPolicy?)namingPolicyField!.GetValue(converter)!;
                string[] names = Enum.GetNames(typeInfo.Type);
                values = [];
                foreach (string name in names)
                {
                    string effectiveName = namingPolicy?.ConvertName(name) ?? name;
                    values.Add((JsonNode)effectiveName);
                }
```

Reply:
@Krzysztof318 I wonder if there is any other option on the table that we can consider for fixing the issue?
   
My concern with the fix is that it will get lost when/if the temporary [JSON schema generator classes](https://github.com/microsoft/semantic-kernel/tree/main/dotnet/src/InternalUtilities/src/Schema) are replaced by ones built into System.Text.Json:
<img width="810" alt="image" src="https://github.com/microsoft/semantic-kernel/assets/68852919/6616791e-6c89-4770-9070-fcd9081518c0">

CC: @eiriktsarpalis

Reply:
Regarding this change in particular, it seems reasonable to also include the type keyword for enums. It might be redundant from the perspective of how `enum` works, but it also doesn't change evaluation semantics fundamentally. If it helps speed things up, you could also try filing a PR to the prototype repo directly.

Reply:
I've created a PR in the prototype repo: https://github.com/eiriktsarpalis/stj-schema-mapper/pull/1

Feel free to update this PR with the diff used there.

Reply:
@eiriktsarpalis @markwallace-microsoft I have updated code to be same like in schema mapper repo.